src/front/routes/Items.tsx

```diff
@@ -35,34 +34,90 @@ const itemsQueryDocument = graphql(`
         }
     }
 `)
+const principles = [
+    "edge",
+    "forge",
+    "grail",
+    "heart",
+    "knock",
+    "lantern",
+    "moon",
+    "moth",
+    "nectar",
+    "rose",
+    "scale",
+    "sky",
+    "winter",
+] as const
 
 type ItemFromQuery = types.ItemsQuery["item"][number]
 
+interface ItemsProps {
+    filters?: {
+        known?: boolean
+        edge?: boolean
+        forge?: boolean
+        grail?: boolean
+        heart?: boolean
+        knock?: boolean
+        lantern?: boolean
+        moon?: boolean
+        moth?: boolean
+        nectar?: boolean
+        rose?: boolean
+        scale?: boolean
+        sky?: boolean
+        winter?: boolean
+        aspect?: string
+    }
```

Comment:
you can do above:

```ts
type Principle = (typeof principles)[number]
```

and then:

```suggestion
    filters?: {
        known?: boolean
        aspect?: string
    } & { [principle: Principle]?: boolean }
```